tests/sentry/api/test_event_search.py

```diff
@@ -561,6 +562,30 @@ def test_blocked_keys(self):
             SearchFilter(key=SearchKey(name="message"), operator="=", value=SearchValue("text")),
         ]
 
+    def test_email_free_text(self):
+        config = SearchConfig.create_from(default_config, allow_free_text_email=True)
+
+        assert parse_search_query(
+            "some_key:123 test.com *@test.com a_1@test.com.uk @name", config=config
+        ) == [
+            SearchFilter(key=SearchKey(name="some_key"), operator="=", value=SearchValue("123")),
+            SearchFilter(
+                key=SearchKey(name="message"), operator="=", value=SearchValue("test.com")
+            ),
+            SearchFilter(
+                key=SearchKey(name="user.email"), operator="=", value=SearchValue("*@test.com")
+            ),
+            SearchFilter(
+                key=SearchKey(name="user.email"), operator="=", value=SearchValue("a_1@test.com.uk")
+            ),
+            SearchFilter(key=SearchKey(name="message"), operator="=", value=SearchValue("@name")),
+        ]
+
+        with pytest.raises(
+            InvalidSearchQuery, match="Please add a relevant tag to search an email"
+        ):
+            assert parse_search_query("some_key:123 test.com @name *@test.com a_1@test.com.uk")
```

Comment:
Ah I think I misunderstood you yesterday... I think in the non-replay case, where `allow_free_text_email = False`, let's preserve the existing behavior, so this should return something like `some_key:123 message:"test.com @name *@test.com a_1@test.com.uk"`